File: 2586-longest-square-streak-in-an-array/longest-square-streak-in-an-array.java
class Solution {
    public int longestSquareStreak(int[] nums) {
        Set<Long>st=new HashSet<>();
        for(int num :nums){
            st.add((long)num);
        }
          int maxi=0;
        for(long num :nums){
        int streak=0;
       while(st.contains(num)){
            streak++;
            num=num*num;
            if(num>1e5)break;
        }
        maxi=Math.max(maxi,streak);
        }
        return (maxi<2)?-1:maxi;
    }
}

File: 2586-longest-square-streak-in-an-array/longest-square-streak-in-an-array.cpp
class Solution {
public:
    int longestSquareStreak(vector<int>& nums) {
        map<double,int>mpp;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++){
             double t=sqrt(nums[i]);
            if(mpp.find(t)!=mpp.end()){
               
                mpp[nums[i]]=mpp[t]+1;
            }
            else{
                mpp[nums[i]*1.0]=1;
            }
        }
        int maxi=1;
        for(auto it :mpp){
            maxi=max(maxi,it.second);
        }
        return (maxi==1)? -1:maxi;
    }
};